### ISSUE TITLE
Crash report: Segfaults in Sierra

### ISSUE THREAD

**Issue by AeroCross** [github/hub#1298]
Hi there!

Whilst using `hub` in several occasions I've had this:

`[1]    34165 segmentation fault  hub fetch`

With `fetch` and other commands.

Here's the latest core dump:

```
Process:               hub [13091]
Path:                  /usr/local/Cellar/hub/2.2.3/bin/hub
Identifier:            hub
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        zsh [12841]
Responsible:           hub [13091]
User ID:               501

Date/Time:             2016-10-06 11:38:06.499 +1100
OS Version:            Mac OS X 10.12 (16A323)
Report Version:        12
Anonymous UUID:        FAEEC505-EE2A-5143-EEE7-C89D8CE7110A

Sleep/Wake UUID:       EC0F63A7-C784-4541-90B7-E7BB5FED4B89

Time Awake Since Boot: 10000 seconds
Time Since Wake:       5400 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x000009eeacfb0ad2
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]

VM Regions Near 0x9eeacfb0ad2:
    VM_ALLOCATE            000000c820100000-0000014820002000 [512.0G] ---/rwx SM=NUL  
--> 
    STACK GUARD            00007fff5bc00000-00007fff5f400000 [ 56.0M] ---/rwx SM=NUL  stack guard for thread 0

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   hub                           	0x000000000000faeb runtime.unlock + 331 (lock_sema.go:107)

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x000009eeacfb0813  rbx: 0x00000000007e85c0  rcx: 0x00000000007e85c0  rdx: 0x000009eeacfb0812
  rdi: 0x00007fff5fbfee40  rsi: 0x0000000000b04000  rbp: 0x000009eeacfb0813  rsp: 0x00007fff5fbfef50
   r8: 0x0000000000000015   r9: 0x000000000000002d  r10: 0x000000007a4d09a2  r11: 0x000000005ba3e4ea
  r12: 0x00000500a80c9aab  r13: 0x000009ee8f3d6580  r14: 0x147aca530effcc00  r15: 0x0000000000000000
  rip: 0x000000000000faeb  rfl: 0x0000000000010206  cr2: 0x000009eeacfb0ad2
  
Logical CPU:     0
Error Code:      0x00000004
Trap Number:     14


Binary Images:
            0x1000 -           0x7c33b7 +hub (???) /usr/local/bin/hub
         0xfa12000 -          0xfa4f1c7  dyld (421.1) <A525EAEA-AF86-30C2-B360-3D093B4F0828> /usr/lib/dyld
    0x7fff95c9b000 -     0x7fff96134fef  com.apple.CoreFoundation (6.9 - 1348.1) <4F7826EC-CC36-38B4-A8F3-DC57BF00EE9B> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
    0x7fff97c59000 -     0x7fff97ceefff  com.apple.framework.IOKit (2.0.2 - 1324.1.7) <10BDCD1E-0463-3C53-BF11-C9A0919C1588> /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
    0x7fff9bf8c000 -     0x7fff9c2a7fff  com.apple.security (7.0 - 57740.1.18) <AF5E42F1-B513-3410-A5BE-1EC9CBAF8452> /System/Library/Frameworks/Security.framework/Versions/A/Security
    0x7fffa9590000 -     0x7fffa9591ff3  libDiagnosticMessagesClient.dylib (102) <422911A4-E273-3E88-BFC4-DF6470E48242> /usr/lib/libDiagnosticMessagesClient.dylib
    0x7fffa97c9000 -     0x7fffa97c9fff  libOpenScriptingUtil.dylib (172) <D025E180-BB3B-3FFA-98FC-B6835354D723> /usr/lib/libOpenScriptingUtil.dylib
    0x7fffa97ce000 -     0x7fffa97cfff3  libSystem.B.dylib (1238) <168B3C56-081B-3998-9A44-681EB4C6828F> /usr/lib/libSystem.B.dylib
    0x7fffa98df000 -     0x7fffa98dfff3  libauto.dylib (187) <5BBF6A00-CC76-389D-84E7-CA88EDADE683> /usr/lib/libauto.dylib
    0x7fffa98e0000 -     0x7fffa98f0ff3  libbsm.0.dylib (34) <20084796-B04D-3B35-A003-EA11459557A9> /usr/lib/libbsm.0.dylib
    0x7fffa98f1000 -     0x7fffa98ffff7  libbz2.1.0.dylib (38) <6FD3B63F-0F86-3A25-BD5B-E243F58792C9> /usr/lib/libbz2.1.0.dylib
    0x7fffa9900000 -     0x7fffa9956ff7  libc++.1.dylib (307.4) <BEE86868-F831-384C-919E-2B286ACFE87C> /usr/lib/libc++.1.dylib
    0x7fffa9957000 -     0x7fffa9981fff  libc++abi.dylib (307.2) <1CEF8ABB-7E6D-3C2F-8E0A-E7884478DD23> /usr/lib/libc++abi.dylib
    0x7fffa99a9000 -     0x7fffa99a9ff7  libcoretls.dylib (121.1.1) <F3BAB54C-DA82-37C3-A218-46FEB179D555> /usr/lib/libcoretls.dylib
    0x7fffa99aa000 -     0x7fffa99abffb  libcoretls_cfhelpers.dylib (121.1.1) <D12709F3-BBE3-3CF7-9070-C2A0803F5526> /usr/lib/libcoretls_cfhelpers.dylib
    0x7fffa9db4000 -     0x7fffa9db4fff  libenergytrace.dylib (15) <A1B040A2-7977-3097-9ADF-34FF181EB970> /usr/lib/libenergytrace.dylib
    0x7fffa9ebd000 -     0x7fffaa0e2fff  libicucore.A.dylib (57131.0.1) <72594943-D25E-3F70-A4AC-8E9F9F442B7E> /usr/lib/libicucore.A.dylib
    0x7fffaa0ea000 -     0x7fffaa103ffb  liblzma.5.dylib (10) <44BD0279-99DD-36B5-8A6E-C11432E2098D> /usr/lib/liblzma.5.dylib
    0x7fffaa470000 -     0x7fffaa840d97  libobjc.A.dylib (706) <F9AFE665-A3A2-3285-9495-19803A565861> /usr/lib/libobjc.A.dylib
    0x7fffaa843000 -     0x7fffaa847fff  libpam.2.dylib (21) <352F8FF6-2248-3594-B357-8D031DEFB06C> /usr/lib/libpam.2.dylib
    0x7fffaa902000 -     0x7fffaaa4afff  libsqlite3.dylib (252) <FCA29CD2-E8E4-3324-8E2C-19BDC11F0338> /usr/lib/libsqlite3.dylib
    0x7fffaab3b000 -     0x7fffaab48fff  libxar.1.dylib (354) <AAFCF321-8100-3622-880B-9661DCA35D69> /usr/lib/libxar.1.dylib
    0x7fffaab4c000 -     0x7fffaac3bffb  libxml2.2.dylib (30.11) <26676155-64CE-323F-B73C-AD77AF5982E8> /usr/lib/libxml2.2.dylib
    0x7fffaac66000 -     0x7fffaac77ff3  libz.1.dylib (67) <46E3FFA2-4328-327A-8D34-A03E20BFFB8E> /usr/lib/libz.1.dylib
    0x7fffaac86000 -     0x7fffaac8aff7  libcache.dylib (79) <84E55656-FDA9-3B29-9E4F-BE31B2C0AA3C> /usr/lib/system/libcache.dylib
    0x7fffaac8b000 -     0x7fffaac95fff  libcommonCrypto.dylib (60092.1.2) <79F738D8-0AD7-3DEA-AF80-E0F8B90B74E3> /usr/lib/system/libcommonCrypto.dylib
    0x7fffaac96000 -     0x7fffaac9dfff  libcompiler_rt.dylib (62) <486BDE52-81B4-3446-BD72-23977CAE556F> /usr/lib/system/libcompiler_rt.dylib
    0x7fffaac9e000 -     0x7fffaaca6fff  libcopyfile.dylib (138) <0DA49B77-56EC-362D-98FF-FA78CFD986D6> /usr/lib/system/libcopyfile.dylib
    0x7fffaaca7000 -     0x7fffaad29ff3  libcorecrypto.dylib (442.1.3) <1CA8B679-810E-3C04-945A-4AC19372CDC7> /usr/lib/system/libcorecrypto.dylib
    0x7fffaad2a000 -     0x7fffaad5cfff  libdispatch.dylib (703.1.4) <12B1D95B-283D-325D-85AB-29A2FFD36987> /usr/lib/system/libdispatch.dylib
    0x7fffaad5d000 -     0x7fffaad62ff3  libdyld.dylib (421.1) <CDFBDC9C-418C-369D-B433-F64B0630E640> /usr/lib/system/libdyld.dylib
    0x7fffaad63000 -     0x7fffaad63ffb  libkeymgr.dylib (28) <09CD7CA6-46D2-3A9F-B9F1-7C4CA5CA0D68> /usr/lib/system/libkeymgr.dylib
    0x7fffaad64000 -     0x7fffaad70ffb  libkxld.dylib (3789.1.32) <48BDBE7F-AAE6-3050-9017-6959051CF16A> /usr/lib/system/libkxld.dylib
    0x7fffaad71000 -     0x7fffaad71fff  liblaunch.dylib (972.1.5) <DC31FDEA-FD81-335E-BAA8-5A7395D20772> /usr/lib/system/liblaunch.dylib
    0x7fffaad72000 -     0x7fffaad77fff  libmacho.dylib (894) <1EAE5ADD-490C-3B1F-9F97-447BA8E0E90F> /usr/lib/system/libmacho.dylib
    0x7fffaad78000 -     0x7fffaad7aff3  libquarantine.dylib (85) <78EF62D8-C890-3FC0-937A-C2FD8CEF8992> /usr/lib/system/libquarantine.dylib
    0x7fffaad7b000 -     0x7fffaad7cffb  libremovefile.dylib (45) <C4FC07FF-ED86-382E-B06F-33C34718080C> /usr/lib/system/libremovefile.dylib
    0x7fffaad7d000 -     0x7fffaad95ff7  libsystem_asl.dylib (349.1.1) <F0987490-8427-367F-B302-A05A7D61FEBF> /usr/lib/system/libsystem_asl.dylib
    0x7fffaad96000 -     0x7fffaad96ff7  libsystem_blocks.dylib (67) <B8C3701D-5A91-3D35-999D-2DC8D5393525> /usr/lib/system/libsystem_blocks.dylib
    0x7fffaad97000 -     0x7fffaae24fe7  libsystem_c.dylib (1158.1.2) <5F260836-48E4-3F57-8553-62D2DA228A1F> /usr/lib/system/libsystem_c.dylib
    0x7fffaae25000 -     0x7fffaae28ffb  libsystem_configuration.dylib (888.1.2) <67BB9D8B-2430-38AD-81A7-F0EC924B2590> /usr/lib/system/libsystem_configuration.dylib
    0x7fffaae29000 -     0x7fffaae2cfff  libsystem_coreservices.dylib (41.1) <11F22E6C-0DCB-3699-A4F0-C99E301E56E9> /usr/lib/system/libsystem_coreservices.dylib
    0x7fffaae2d000 -     0x7fffaae45ffb  libsystem_coretls.dylib (121.1.1) <8F7E9B12-400D-3276-A9C5-4546B0258554> /usr/lib/system/libsystem_coretls.dylib
    0x7fffaae46000 -     0x7fffaae4cfff  libsystem_dnssd.dylib (765.1.2) <C5FF2025-C60B-39C6-B205-6BF1BC51D1B3> /usr/lib/system/libsystem_dnssd.dylib
    0x7fffaae4d000 -     0x7fffaae76fff  libsystem_info.dylib (503) <C686B834-5E7D-382C-AF6E-44AB78EE83E2> /usr/lib/system/libsystem_info.dylib
    0x7fffaae77000 -     0x7fffaae99ff7  libsystem_kernel.dylib (3789.1.32) <5C68A0D7-C3C9-3E52-B983-EDE9A28AB6FC> /usr/lib/system/libsystem_kernel.dylib
    0x7fffaae9a000 -     0x7fffaaee1fe7  libsystem_m.dylib (3121.4) <E3370D16-EBAA-3C7F-AC56-2D6EAD7DB0A4> /usr/lib/system/libsystem_m.dylib
    0x7fffaaee2000 -     0x7fffaaf00ff7  libsystem_malloc.dylib (116) <3DD17B88-B7A4-38B9-9E95-AB88E1C3B647> /usr/lib/system/libsystem_malloc.dylib
    0x7fffaaf01000 -     0x7fffaaf57fff  libsystem_network.dylib (856.1.8) <A8973360-956A-33BF-9971-14D59C84E0D1> /usr/lib/system/libsystem_network.dylib
    0x7fffaaf58000 -     0x7fffaaf61ff3  libsystem_networkextension.dylib (563.1.11) <1C551832-9360-36DD-A7BA-52B55A171720> /usr/lib/system/libsystem_networkextension.dylib
    0x7fffaaf62000 -     0x7fffaaf6bffb  libsystem_notify.dylib (165) <AF77D471-6B13-36BA-B144-7E7DDB9DBA9F> /usr/lib/system/libsystem_notify.dylib
    0x7fffaaf6c000 -     0x7fffaaf74fe7  libsystem_platform.dylib (126.1.2) <884DDF42-3CAE-334A-82CE-965617130FB1> /usr/lib/system/libsystem_platform.dylib
    0x7fffaaf75000 -     0x7fffaaf7ffff  libsystem_pthread.dylib (218.1.3) <050AE77B-4F4B-334A-A5BA-CF0D10AF5304> /usr/lib/system/libsystem_pthread.dylib
    0x7fffaaf80000 -     0x7fffaaf83fff  libsystem_sandbox.dylib (592.1.3) <90FEF628-316D-3F84-9EF4-F7F510A4FBBB> /usr/lib/system/libsystem_sandbox.dylib
    0x7fffaaf84000 -     0x7fffaaf85fff  libsystem_secinit.dylib (24) <A54B8FEF-E792-3C54-8E0B-E80A376662F2> /usr/lib/system/libsystem_secinit.dylib
    0x7fffaaf86000 -     0x7fffaaf8dfff  libsystem_symptoms.dylib (532.1.1) <B26F656E-94F9-3834-9B03-51C4FF11D1BF> /usr/lib/system/libsystem_symptoms.dylib
    0x7fffaaf8e000 -     0x7fffaafaeff7  libsystem_trace.dylib (518.1.16) <C3CDF7BC-CA3D-34F5-ADF8-46AAAB0B47F8> /usr/lib/system/libsystem_trace.dylib
    0x7fffaafaf000 -     0x7fffaafb4ffb  libunwind.dylib (35.3) <9F7C2AD8-A9A7-3DE4-828D-B0F0F166AAA0> /usr/lib/system/libunwind.dylib
    0x7fffaafb5000 -     0x7fffaafdeff7  libxpc.dylib (972.1.5) <2A901937-48E1-3BF2-83F1-2431156D82B7> /usr/lib/system/libxpc.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 7533
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=140.6M resident=0K(0%) swapped_out_or_unallocated=140.6M(100%)
Writable regions: Total=27.9M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=27.9M(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Kernel Alloc Once                    8K        2 
MALLOC                            18.2M        8 
MALLOC guard page                   16K        4 
STACK GUARD                       56.0M        2 
Stack                             8192K        2 
VM_ALLOCATE                      544.5G        7 
VM_ALLOCATE (reserved)            1056K        2         reserved VM address space (unallocated)
__DATA                            3584K       61 
__LINKEDIT                       111.8M        4 
__TEXT                            28.8M       60 
__UNICODE                          556K        2 
shared memory                        8K        3 
===========                     =======  ======= 
TOTAL                            544.7G      145 
TOTAL, minus reserved VM space   544.7G      145 
```

Is this something you're aware of? Using:

```
$ hub --version
git version 2.9.0
hub version 2.2.3
```

And macOS Sierra.

**Comment by mislav** [comment]
Hi, thanks for reporting, and please install the latest version of hub from Homebrew!

```
brew up && brew upgrade hub
```

The newer versions of hub are built with a version of Go that supports Sierra. Versions of Go older than 1.6.3 didn't support Sierra fully, leading to crashes like these.

**Comment by mikermcneil** [comment]
In case it helps anyone else: After the High Sierra upgrade, I was seeing:

![image](https://user-images.githubusercontent.com/618009/35203215-da07d456-feec-11e7-8e3f-11d28d60444a.png)

I tried the above advice (thanks @mislav!) which mostly worked- though I had to make a few tweaks.  When I tried to do `brew upgrade hub`, I saw:

![image](https://user-images.githubusercontent.com/618009/35203267-0ed3659c-feed-11e7-96a1-62497a48c7ab.png)


Not sure why that was, but it was pretty easy to resolve: just reinstall `hub`:

```
brew install hub
```

And then if you see:

![image](https://user-images.githubusercontent.com/618009/35203295-2dab57ea-feed-11e7-8e40-bfc8dee58953.png)

Just do:

![image](https://user-images.githubusercontent.com/618009/35203346-5e1be106-feed-11e7-8c51-822c15cfdb0f.png)

Then, after reloading my bash profile (e.g. opening a new terminal tab or running `source ~/.profile`), everything worked again as expected!
